fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ReorderableList.kt

```diff
@@ -0,0 +1,270 @@
+            val overscroll = when {
+                draggingItemCumulatedOffset > 0 ->
+                    (endOffset - listState.layoutInfo.viewportEndOffset).coerceAtLeast(0f)
+
+                draggingItemCumulatedOffset < 0 ->
+                    (startOffset - listState.layoutInfo.viewportStartOffset).coerceAtMost(0f)
+
+                else -> 0f
+            }
+            if (overscroll != 0f) {
+                scope.launch {
+                    listState.scrollBy(overscroll)
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Container for draggable list item
+ *
+ * @param state List reordering state
+ * @param key Key of the item to be displayed
+ * @param content Content of the item to be displayed
+ */
+@ExperimentalFoundationApi
+@Composable
+fun LazyItemScope.DragItemContainer(
+    state: ListReorderState,
+    key: Any,
+    content: @Composable () -> Unit,
+) {
+    val modifier = when (key) {
+        state.draggingItemKey -> {
+            Modifier
+                .zIndex(1f)
+                .graphicsLayer {
+                    translationY = state.draggingItemOffset
+                }
+        }
+
+        state.previousKeyOfDraggedItem -> {
+            Modifier
+                .zIndex(1f)
+                .graphicsLayer {
+                    translationY = state.previousItemOffset.value
+                }
+        }
+
+        else -> {
+            Modifier
+                .zIndex(0f)
+                .animateItemPlacement(tween())
+        }
+    }
+    Box(modifier = modifier, propagateMinConstraints = true) {
+        content()
+    }
+}
+
+/**
+ * Calculates the offset of an item taking its height into account
+ */
+private val LazyListItemInfo.endOffset: Int
+    get() = offset + size
+
+/**
+ * Find item based on position on screen
+ *
+ * @param offset Position on screen used to find the item
+ */
+private fun LazyListState.findItem(offset: Float) =
+    layoutInfo.visibleItemsInfo.firstOrNull { item ->
+        offset.toInt() in item.offset..item.endOffset
+    }
+
+/**
+ * Detects press, long press and drag gestures.
+ *
+ * @param listState State of the list
+ * @param reorderState List reordering state used for dragging callbacks
+ * @param shouldLongPressToDrag Whether or not an item should be long pressed to start the dragging gesture
+ */
+fun Modifier.detectVerticalPressAndDrag(
+    listState: LazyListState,
+    reorderState: ListReorderState,
+    shouldLongPressToDrag: Boolean,
+): Modifier = pointerInput(listState, shouldLongPressToDrag) {
+    if (shouldLongPressToDrag) {
+        detectDragGesturesAfterLongPress(
+            onDragStart = { offset -> reorderState.onDragStart(offset.y, true) },
+            onDrag = { change, dragAmount ->
+                change.consume()
+                reorderState.onDrag(dragAmount.y)
+            },
+            onDragEnd = reorderState::onDragInterrupted,
+            onDragCancel = reorderState::onDragInterrupted,
+        )
+    }
```

Comment:
Adding the similar `detectPressAndDragGestures` to `ReorderableList` would make dragging possible without long press first when in selection mode. From my testing on Fenix Release, when list view and selection mode are enabled at the same time, dragging can only start after long press. 

I think this makes sense since the only supported gesture in list view is vertical scroll and we wouldn't want to keep the user stuck/having to reorder a tab to be able to select the first and last tab when many tabs are opened.

Reply:
I think that makes sense! For context, I was comparing the grid and list implementations as I was doing this review pass and checking for any differences and added comments for anything I wanted to learn more about.